scripts/estimate-builds.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# pylint: disable=invalid-name
+
+from collections import defaultdict
+import datetime
+from pathlib import Path
+import sys
+
+import croniter
+
+# Add the root of the repo to PYTHONPATH for utils
+ci_root = Path(__file__).resolve().parents[1]
+sys.path.insert(0, str(ci_root))
+# pylint: disable-next=wrong-import-position
+from utils import get_config_from_generator  # noqa: E402
+
+config = get_config_from_generator()
+
+now = datetime.datetime.now(tz=datetime.timezone.utc)
+week_from_now = now + datetime.timedelta(weeks=1)
+
+builds_per_tree = defaultdict(lambda: defaultdict(lambda: 0))
+for tree in config['tree_schedules']:
+    tree_name = tree['name']
+    tree_llvm_ver = tree['llvm_version']
+
+    # Calculate the number of times that a workflow runs in a week based on its
+    # schedule
+    num_runs = len(
+        list(croniter.croniter_range(now, week_from_now, tree['schedule'])))
+    for build in config['builds']:
+        if tree['git_repo'] == build['git_repo'] and \
+           tree['git_ref'] == build['git_ref'] and \
+           tree_llvm_ver == build['llvm_version']:
+            builds_per_tree[tree_name]['total'] += num_runs
+            builds_per_tree[tree_name][tree_llvm_ver] += num_runs
+
+print(
+    f"Total builds per week: {sum(item['total'] for item in builds_per_tree.values())}"
+)
```

Comment:
does this no longer fit on one line?

Reply:
No, `yapf` will force this because the line is over 80 characters. I could do this:

```diff
diff --git a/scripts/estimate-builds.py b/scripts/estimate-builds.py
index a545768..3ac188a 100755
--- a/scripts/estimate-builds.py
+++ b/scripts/estimate-builds.py
@@ -35,9 +35,8 @@ for tree in config['tree_schedules']:
             builds_per_tree[tree_name]['total'] += num_runs
             builds_per_tree[tree_name][tree_llvm_ver] += num_runs

-print(
-    f"Total builds per week: {sum(item['total'] for item in builds_per_tree.values())}"
-)
+total_builds = sum(item['total'] for item in builds_per_tree.values())
+print(f"Total builds per week: {total_builds}")

 # Sort the list of builds by total number of builds descending
 for tree, builds in sorted(builds_per_tree.items(),
```

or perhaps we bump the line length to 100 like the kernel, it is not like we do not have the screen real estate nowadays but that is obviously a larger change for the whole repo. Up to you.

Reply:
I prefer that diff, please apply it.

Reply:
Done: 4dde30ae0a0be9bf5671d7fc6106c882fd610cff